features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/attachments/preview/AttachmentsPreviewPresenter.kt

```diff
@@ -92,13 +94,22 @@ class AttachmentsPreviewPresenter @AssistedInject constructor(
                         // Pre-processing is done, send the attachment
                         val caption = markdownTextEditorState.getMessageMarkdown(permalinkBuilder)
                             .takeIf { it.isNotEmpty() }
-                        ongoingSendAttachmentJob.value = coroutineScope.launch {
+
+                        // Send it using the session coroutine scope so it doesn't matter if this screen or the chat one is closed
+                        val sendOnBackground = featureFlagService.isFeatureEnabled(FeatureFlags.MediaUploadOnSendQueue)
+                        val sendMediaScope = if (sendOnBackground) sessionCoroutineScope else coroutineScope
+                        ongoingSendAttachmentJob.value = sendMediaScope.launch {
                             sendPreProcessedMedia(
                                 mediaUploadInfo = mediaUploadInfo.data,
                                 caption = caption,
                                 sendActionState = sendActionState,
                             )
                         }
+
```

Comment:
Just a remark here: when the send queue is enabled, the duration of the invokation of `sendPreProcessedMedia` stay quite short (but not negligible). I have measure that it can take up to 1s for big files. The processing of the thumbnail creation and video shrinking is much longer. So this PR is OK, but the gain is not that big.

Also note that if the user close the screen during the media pre-processing, it will cancel the media sending as well. So maybe use the session scope also for the media pre-processing? In this case the local echo will take longer to appear and killing the app during the processing will stop the sending.